src/OmniSharp.Roslyn.CSharp/Services/Intellisense/CompletionItemExtensions.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Reflection;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Completion;
+
+namespace OmniSharp.Roslyn.CSharp.Services.Intellisense
+{
+    internal static class CompletionItemExtensions
+    {
+        private static MethodInfo _getSymbolsAsync;
+
+        static CompletionItemExtensions()
+        {
+            var symbolCompletionItemType = typeof(CompletionItem).GetTypeInfo().Assembly.GetType("Microsoft.CodeAnalysis.Completion.Providers.SymbolCompletionItem");
+            _getSymbolsAsync = symbolCompletionItemType.GetMethod("GetSymbolsAsync", BindingFlags.Public | BindingFlags.Static);
+        }
+
+        public static async Task<IEnumerable<ISymbol>> GetCompletionSymbols(this CompletionItem completionItem, IEnumerable<ISymbol> recommendedSymbols, Document document)
+        {
+            // for SymbolCompletionProvider, use the logic of extracting information from recommended symbols
```

Comment:
I kind of hate that we pre-compute recommendedSymbols (which is essentially like calling the CompletionService _twice_) just to handle this special case.